backend/cli_commands/AddFolderCommand.py

```diff
@@ -27,29 +26,41 @@ def command(self, args):
         add_mission_from_folder(args.path, args.location, args.notes)
 
 
-storage = DefaultStorage()
+storage = File.file.field.storage
 
 
 def get_duration(path):
-    metadata = load_yaml_metadata(path)
-    duration = get_mcap_duration_from_yaml(metadata)
-    return duration
-
-
-def load_yaml_metadata(yaml_filepath):
-    """Load YAML metadata."""
-    with storage.open(yaml_filepath, "r") as file:
-        return yaml.safe_load(file)
-
-
-def get_mcap_duration_from_yaml(metadata):
-    """Extract duration in seconds from YAML."""
-    nanoseconds = (
-        metadata.get("rosbag2_bagfile_information", {})
-        .get("duration", {})
-        .get("nanoseconds", 0)
-    )
-    return nanoseconds / 1e9
+    with storage.open(path, "rb") as f:
+        reader = make_reader(f)
+        statistics = reader.get_summary().statistics
+        return (statistics.message_end_time - statistics.message_start_time) / 1**-9
```

Comment:
```suggestion
        return (statistics.message_end_time - statistics.message_start_time) * 10**-9
```
You are just dividing by one here. You need to do it like this to get seconds